build: krb-related errors

Hi developers, When i trying build cockroachdb from source, it will fail when build krb.

**To Reproduce**
Follow the instruction in https://wiki.crdb.io/wiki/spaces/CRDB/pages/181338446/Getting+and+building+from+source, trying to build cockroachdb, but it fail when linking kdb5_util

1.mkdir -p $(go env GOPATH)/src/github.com/cockroachdb
2.cd $(go env GOPATH)/src/github.com/cockroachdb
3.git clone https://github.com/cockroachdb/cockroach
4.cd cockroach
5.make buildshort

**Expected behavior**
Build success

**Additional data / screenshots**
Here is part of the build output

making all in kadmin/dbutil...
gcc -L../../lib -Wl,--enable-new-dtags -Wl,-rpath -Wl,/usr/local/lib -g -O2   -o kdb5_util kdb5_util.o kdb5_create.o kadm5_create.o kdb5_destroy.o kdb5_stash.o import_err.o strtok.o dump.o ovload.o kdb5_mkey.o tabdump.o tdumputil.o ../cli/getdate.o -lkadm5srv_mit  -lkdb5 -lkrb5_db2 -lkadm5srv_mit -lgssrpc -lgssapi_krb5 -lgssrpc -lgssapi_krb5 -ldl -pthread  -lkrb5 -lk5crypto -lcom_err -lkrb5support  -lkeyutils -lresolv  -ldl
/usr/bin/ld: ../../lib/libkadm5srv_mit.a(server_kdb.o):/home/rebellion/go/src/github.com/cockroachdb/cockroach/c-deps/krb5/src/lib/kadm5/srv/server_kdb.c:18: multiple definition of `master_keyblock'; kdb5_util.o:(.bss+0x40): first defined here
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:663: kdb5_util] Error 1
make[2]: *** [Makefile:853: all-recurse] Error 1
make[1]: *** [Makefile:1533: all-recurse] Error 1
make: *** [Makefile:776: /home/rebellion/go/native/x86_64-pc-linux-gnu/krb5/lib/libgssapi_krb5.a] Error 2
make: *** Deleting file '/home/rebellion/go/native/x86_64-pc-linux-gnu/krb5/lib/libgssapi_krb5.a'

**Environment:**
 - CockroachDB version 20.1.1, master branch, commit id 021d87644837dfc7e7cd56af386f94b93f41623e
 - Server OS: Arch Linux version 5.6.14-arch1-1 (linux@archlinux) (gcc version 10.1.0 (GCC))
 - Client app: None

**Additional context**

Yeah, the output is

Makefile:52: build/defs.mk: No such file or directory
GOPATH set to /go
bash: build/defs.mk.sig: Permission denied
Makefile:812: recipe for target 'build/defs.mk.sig' failed
make: *** [build/defs.mk.sig] Error 1

I'm running into the same after upgrading to Ubuntu 20.10 (gcc 10.2). Used to work on Ubuntu 20.04.

```
gcc -L../../lib -Wl,--enable-new-dtags -Wl,-rpath -Wl,/usr/local/lib -g -O2   -o kdb5_util kdb5_util.o kdb5_create.o kadm5_create.o kdb5_destroy.o kdb5_stash.o import_err.o strtok.o dump.o ovload.o kdb5_mkey.o tabdump.o tdumputil.o ../cli/getdate.o -lkadm5srv_mit  -lkdb5 -lkrb5_db2 -lkadm5srv_mit -lgssrpc -lgssapi_krb5 -lgssrpc -lgssapi_krb5 -ldl -pthread  -lkrb5 -lk5crypto -lcom_err -lkrb5support  -lresolv  -ldl
/usr/bin/ld: ../../lib/libkadm5srv_mit.a(server_kdb.o):/home/andrei/work/src/github.com/cockroachdb/cockroach/c-deps/krb5/src/lib/kadm5/srv/server_kdb.c:18: multiple definition of `master_keyblock'; kdb5_util.o:/home/andrei/work/src/github.com/cockroachdb/cockroach/c-deps/krb5/src/kadmin/dbutil/kdb5_util.c:109: first defined here
```

This has got to be a gcc/ld version thing, I'm thinking.
@ajwerner as another Ubuntu user, did you upgrade to 20.10? Can you confirm?

--- a/Makefile
+++ b/Makefile
@@ -625,8 +625,9 @@ $(KRB5_DIR)/Makefile: $(C_DEPS_DIR)/krb5-rebuild $(KRB5_SRC_DIR)/src/configure
 	mkdir -p $(KRB5_DIR)
 	@# NOTE: If you change the configure flags below, bump the version in
 	@# $(C_DEPS_DIR)/krb5-rebuild. See above for rationale.
-	@# If CFLAGS is set to -g1 then make will fail. Use "env -" to clear the environment.
-	cd $(KRB5_DIR) && env -u CFLAGS -u CXXFLAGS $(KRB5_SRC_DIR)/src/configure $(xconfigure-flags) --enable-static --disable-shared
+	@# If CFLAGS is set to -g1 then make will fail.
+	@# We specify -fcommon to get around duplicate definition errors in recent gcc.
+	cd $(KRB5_DIR) && env -u CXXFLAGS CFLAGS="-fcommon"  $(KRB5_SRC_DIR)/src/configure $(xconfigure-flags) --enable-static --disable-shared
 
 $(PROTOBUF_DIR)/Makefile: $(C_DEPS_DIR)/protobuf-rebuild | bin/.submodules-initialized
 	rm -rf $(PROTOBUF_DIR)

--- a/c-deps/krb5-rebuild
+++ b/c-deps/krb5-rebuild
@@ -1,4 +1,4 @@
 Bump the version below when changing krb5 configure flags. Search for "BUILD
 ARTIFACT CACHING" in build/common.mk for rationale.
 
-2
+3

